keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourcePersister.kt

```diff
@@ -71,6 +73,15 @@ class ResourcePersister(
         )
       }
     )
+
+    try {
+      validate(new)
+    } catch (e: ValidationException) {
+      log.warn("Validation of ${new.name} failed, deleting delivery config")
+      deleteDeliveryConfig(new.name)
```

Comment:
Why do we need to delete it? We didn't save it yet.

Reply:
Ah you're right, we only need to delete the resources. Good catch.

Reply:
I would propose we don't persist the resources until after this validation check

Reply:
That's definitely better. I'll do that

Reply:
Fixed. Wow much better. So wise.